File: Sesion 9/Actividades/Actividad 4/actividad4.cpp
#include <iostream>
using namespace std;

class Animal{
    public:
        virtual void hacerSonido() = 0;
};

class Perro : public Animal{
    public:
        void hacerSonido() override{
            cout << "Wau Wau" << endl;
        }
};

class Gato : public Animal{
    public:
        void hacerSonido() override{
            cout << "Miau" << endl;
        }
};

class Vaca : public Animal{
    public:
        void hacerSonido() override{
            cout << "Muuu" << endl;
        }
};

int main(){
    Animal* array_animales[10];
    Perro *perrito = new Perro();
    Gato *gatito = new Gato();
    Vaca *vaquita = new Vaca();

    array_animales[0] = perrito;
    array_animales[1] = gatito;
    array_animales[2] = vaquita;

    for (int i = 0; i < 3; i++) {
        array_animales[i]->hacerSonido();
    }
}

File: Sesion 9/Actividades/Actividad 4/actividad4.java
abstract class Animal {
    public abstract void hacerSonido();
}

class Perro extends Animal {
    @Override
    public void hacerSonido() {
        System.out.println("Wau Wau");
    }
}

class Gato extends Animal {
    @Override
    public void hacerSonido() {
        System.out.println("Miau");
    }
}

class Vaca extends Animal {
    @Override
    public void hacerSonido() {
        System.out.println("Muuu");
    }
}

public class actividad4 {
    public static void main(String[] args) {
        Animal[] arrayAnimales = new Animal[10];
        Perro perrito = new Perro();
        Gato gatito = new Gato();
        Vaca vaquita = new Vaca();

        arrayAnimales[0] = perrito;
        arrayAnimales[1] = gatito;
        arrayAnimales[2] = vaquita;

        for (int i = 0; i < 3; i++) {
            arrayAnimales[i].hacerSonido();
        }
    }
}
